src/Npgsql/ConnectorPool.Multiplexing.cs

```diff
@@ -188,14 +193,9 @@ async Task MultiplexingWriteLoop()
                     }
                     else
                     {
-                        // We reuse the timeout's cancellation token source as long as it hasn't fired, but once it has
-                        // there's no way to reset it (see https://github.com/dotnet/runtime/issues/4694)
-                        var timeoutTimeSpan = TimeSpan.FromTicks(timeout);
-                        timeoutTokenSource.CancelAfter(timeoutTimeSpan);
-                        if (timeoutTokenSource.IsCancellationRequested)
+                        if (timeoutTokenSource != null)
                         {
-                            timeoutTokenSource.Dispose();
-                            timeoutTokenSource = new CancellationTokenSource(timeoutTimeSpan);
+                            timeoutTokenSource.Start();
                             timeoutToken = timeoutTokenSource.Token;
```

Comment:
This call site then will be simplified to:
```csharp
timeoutToken = timeoutTokenSource.Start();
```

Reply:
You mean that `Start()` should return the token?
Good idea!